AllReadyApp/Web-App/AllReady/Features/Notifications/VolunteerAssignedToItineraryHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Threading.Tasks;
+using AllReady.Models;
+using MediatR;
+using Microsoft.Data.Entity;
+
+namespace AllReady.Features.Notifications
+{
+    public class VolunteerAssignedToItineraryHandler : IAsyncNotificationHandler<VolunteerAssignedToItinerary>
+    {
+        private readonly AllReadyContext _context;
+        private readonly IMediator _mediator;
+
+        public VolunteerAssignedToItineraryHandler(AllReadyContext context, IMediator mediator)
+        {
+            _context = context;
+            _mediator = mediator;
+        }
+
+        public async Task Handle(VolunteerAssignedToItinerary notification)
+        {
+            var taskSignup = await _context.TaskSignups
+                .AsNoTracking()
+                .Include(x => x.User)
+                .SingleAsync(ts => ts.Id == notification.TaskSignupId)
+                .ConfigureAwait(false);
```

Comment:
@dpaquette, here are a couple links that helped me to get my head around it:
- http://stackoverflow.com/questions/13489065/best-practice-to-call-configureawait-for-all-server-side-code
- http://blog.stephencleary.com/2012/02/async-and-await.html
- http://blog.stephencleary.com/2012/07/dont-block-on-async-code.html

But Sean is really the one who helped me understand it best. He basically said (and I'm paraphrasing): 
"on any invocations of an async method that you await, if the place the method is being invoked from is NOT a UI-bound thread (MVC/WPF/Wndows Forms, etc...), use .ConfigureAwait(false)"